src/game-logic/reducers/applyCrows.js

```diff
@@ -1,39 +1,53 @@
-import { itemType } from '../../enums'
-import { itemsMap } from '../../data/maps'
-import { getPlotContentType } from '../../utils'
-import { CROW_CHANCE } from '../../constants'
-import { CROW_ATTACKED } from '../../templates'
+import { doesPlotContainCrop } from '../../utils'
+import { CROW_CHANCE, MAX_CROWS } from '../../constants'
+import { CROWS_DESTROYED } from '../../templates'
 
-import { fieldHasScarecrow, updateField } from './helpers'
+import { modifyFieldPlotAt } from './modifyFieldPlotAt'
+
+import { fieldHasScarecrow } from './helpers'
 
 // TODO: Add tests for this reducer.
 /**
  * @param {farmhand.state} state
  * @returns {farmhand.state}
  */
 export const applyCrows = state => {
-  const { field } = state
+  const { field, purchasedField } = state
+
+  if (fieldHasScarecrow(field) || Math.random() > CROW_CHANCE) {
```

Comment:
Would it make sense to use `isRandomNumberLessThan()` instead of the `Math.random()` check here?

Reply:
yeah, i think so, though it's going to be `!isRandomNumberLessThan(CROW_CHANCE)` which for some reason feels awkward to me. i'm going to make the change for now but curious if you think that makes sense, or maybe the value for `CROW_CHANCE` should be changed to `.6` instead of `.4` though that doesn't seem right to me either.

Reply:
> it's going to be !isRandomNumberLessThan(CROW_CHANCE) which for some reason feels awkward to me.

I don't find it overly awkward, personally. In general, I try to follow [this principle](https://libquotes.com/eric-s-raymond/quote/lbt5w7w):

> Smart data structures and dumb code works a lot better than the other way around.

In terms of this situation, it means that the data (or at least Farmhand's `CONSTANT` values) should express the rules of the game as clearly as possible, even at the some expense of straightforwardness in the logic. There are exceptions to this of course, but I think it results in a more cohesive system overall.

That isn't a strongly-held opinion, it's just how I think about this stuff. I trust your judgment, so feel free to change `0.4` to `0.6` and drop the negation if you think that makes more sense. The more important thing here is using consistent solutions for similar problems (using `isRandomNumberLessThan` for dice rolls). 🙂

Reply:
that all makes sense. i decided to go with `isRandomNumberLessThan(1 - CROW_CHANCE)` ultimately. let me know what you think!

Reply:
Oh whoops, I just saw this. I think that's a good approach! :+1: